Missing max_atomic_width in avr spec.

(Moving this here from https://github.com/avr-rust/rust-legacy-fork/issues/172)

The `avr` target now specifies no `max_atomic_width` in the [target spec file](https://github.com/rust-lang/rust/blob/master/compiler/rustc_target/src/spec/avr_gnu_base.rs), which means it [falls back to the pointer size](https://github.com/rust-lang/rust/blob/d7f94516345a36ddfcd68cbdf1df835d356795c3/src/librustc_target/spec/mod.rs#L1121-L1123) of 16 bits:

```rust
    let target = rustc_target::spec::TargetTriple::from_triple("avr-unknown-gnu-atmega328");
    println!("{}", rustc_target::spec::Target::search(&target).unwrap().max_atomic_width());
    // output: 16
```

This seems incorrect. As far as I know, AVR does not have any atomic operations (especially not 16 bit).

Compiling something for AVR that uses atomics results in linker errors: `undefined reference to __sync_val_compare_and_swap_1` `undefined reference to __sync_val_compare_and_swap_2`, etc.

--- a/compiler/rustc_target/src/spec/avr_gnu_base.rs
+++ b/compiler/rustc_target/src/spec/avr_gnu_base.rs
@@ -45,6 +45,8 @@ pub fn target(target_cpu: String) -> TargetResult {
             late_link_args: vec![(LinkerFlavor::Gcc, vec!["-lgcc".to_owned()])]
                 .into_iter()
                 .collect(),
+            max_atomic_width: Some(0),
+            atomic_cas: false,
             ..TargetOptions::default()
         },
     })

